tests/testthat/test-Samples-methods.R

```diff
@@ -1,53 +1,1440 @@
+  samples <- h_as_samples(list(alpha0 = c(0, -1, 1, 2), alpha1 = c(0, 2, 1, -1)))
+  samples2 <- h_as_samples(
+    list(alpha0 = seq(from = 1, length.out = 50), alpha1 = seq(from = 60, length.out = 50))
+  )
+  expect_identical(size(samples), 4L)
+  expect_identical(size(samples2), 50L)
+})
+
+# names ----
+
+## Samples ----
+
+test_that("names-Samples returns correct names of the parameters", {
+  samples <- h_as_samples(list(alpha0 = c(0, -1), alpha1 = c(2, 1), beta = c(4, 7)))
+  samples2 <- h_as_samples(
+    list(a = matrix(5:8, nrow = 2), z = matrix(1:4, nrow = 2))
+  )
+  expect_identical(names(samples), c("alpha0", "alpha1", "beta"))
+  expect_identical(names(samples2), c("a", "z"))
+})
+
+# get ----
+
+## Samples ----
+
+test_that("get-Samples fails gracefully with bad input", {
+  samples <- Samples(data = list(good = 1:3), options = McmcOptions(samples = 3))
+  expect_error(
+    get(samples, "bad"),
+    "Assertion on 'pos' failed: Must be element of set \\{'good'\\}, but is 'bad'."
+  )
+  expect_error(
+    get(samples, c("bad", "worse")),
+    "Assertion on 'pos' failed: Must have length 1."
+  )
+
+  dualSamples <- Samples(data = list(good = matrix(1:6, ncol = 2)), options = McmcOptions(samples = 3))
+  expect_error(
+    get(dualSamples, "good", envir = "NotNumeric"),
+    "Assertion on 'envir' failed: Must be of type 'integer', not 'character'."
+  )
+  expect_error(
+    get(dualSamples, "good", envir = pi),
+    "Assertion on 'envir' failed: Must be of type 'integer', not 'double'."
+  )
+  expect_error(
+    get(dualSamples, "good", envir = 99L),
+    "Assertion on 'envir' failed: Must be a subset of \\{'1','2'\\}, but has additional elements \\{'99'\\}"
+  )
+})
+
+test_that("get-Samples returns correct values", {
+  mcmcOptions <- McmcOptions(samples = 3)
+  samples <- Samples(data = list(alpha0 = 1:3, alpha1 = 4:6), options = mcmcOptions)
+
+  for (param in names(samples@data)) {
+    expected <- data.frame(
+      Iteration = as.integer(1:3),
+      Chain = 1L,
+      Parameter = as.factor(param),
+      value = as.double(samples@data[[param]])
+    )
+
+    attr(expected, "description") <- param
+    attr(expected, "nBurnin") <- mcmcOptions@burnin
+    attr(expected, "nChains") <- 1L
+    attr(expected, "nParameters") <- 1L
+    attr(expected, "nThin") <- mcmcOptions@step
+    attr(expected, "nIterations") <- mcmcOptions@iterations
+    attr(expected, "parallel") <- FALSE
+
+    expect_identical(get(samples, param), expected)
+  }
+
+  dualData <- DataDual(doseGrid = c(seq(from = 10, to = 80, by = 10)))
+  dualModel <- DualEndpointRW(
+    mean = c(0, 1),
+    cov = matrix(c(1, 0, 0, 1), nrow = 2),
+    sigma2betaW = 0.01,
+    sigma2W = c(a = 0.1, b = 0.1),
+    rho = c(a = 1, b = 1),
+    rw1 = TRUE
+  )
+  mcmcOptions <- McmcOptions(burnin = 5, step = 2, samples = 2)
+  set.seed(94)
+  dualSamples <- mcmc(dualData, dualModel, mcmcOptions)
+  for (param in c("betaZ")) { # names(dualSamples@data)) {
```

Comment:
generally we don't want commented code - not sure if that was disabled now with the linter options?
additionally, we normally don't use a loop if there is only a single iteration, as here

Reply:
Fixed.  Commented code should still be flagged by `lintr`.  I only disabled the checks for whitespace at the end of lines and blank lines at the end of files.